### DISCUSSION TITLE
enable ldap to a zimbra mail server

### DISCUSSION THREAD

**Discussion by cubanlinux** [jsuto/piler#197] [Q&A] [answered]
hey 

i recently run piler on docker but when i tried connect whit my zimbra email server
I don't know how to put the necessary configuration inside the container to connect with my Zimbra mail server

this is my docker-compose.yml file

services:
  mysql:
    image: mariadb:11.1.2
    container_name: mysql
    restart: unless-stopped
    cap_drop:
      - ALL
    cap_add:
      - dac_override
      - setuid
      - setgid
    environment:
      - MYSQL_DATABASE=piler
      - MYSQL_USER=piler
      - MYSQL_PASSWORD=piler123
      - MYSQL_RANDOM_ROOT_PASSWORD=yes
    command: --character-set-server=utf8mb4 --collation-server=utf8mb4_unicode_ci
    volumes:
      - db_data:/var/lib/mysql
      - ./piler.cnf:/etc/mysql/conf.d/piler.cnf:ro

  manticore:
    image: manticoresearch/manticore:6.3.2
    container_name: manticore
    restart: unless-stopped
    volumes:
      - ./manticore.conf:/etc/manticoresearch/manticore.conf
      - piler_manticore:/var/lib/manticore
    deploy:
      resources:
        reservations:
          memory: 512M
        limits:
          memory: 512M

  piler:
    image: sutoj/piler:1.4.6
    container_name: piler
    init: true
    environment:
      - MANTICORE_HOSTNAME=manticore
      - MEMCACHED_HOSTNAME=memcached
      - MYSQL_HOSTNAME=mysql
      - MYSQL_DATABASE=piler
      - MYSQL_USER=piler
      - MYSQL_PASSWORD=piler123
      - PILER_HOSTNAME=piler.onei.gob.cu
      - RT=1
    ports:
      - "25:25"
      - "80:80"
    volumes:
      - piler_etc:/etc/piler
      - piler_store:/var/piler/store
    healthcheck:
      test: curl -s smtp://localhost/
      interval: "20s"
      timeout: "3s"
      start_period: "15s"
      retries: 3
    deploy:
      resources:
        reservations:
          memory: 512M
        limits:
          memory: 512M
    depends_on:
      - "mysql"
      - "memcached"
      - "manticore"

  memcached:
    image: memcached
    container_name: memcached
    command: -m 64

volumes:
  db_data: {}
  piler_etc: {}
  piler_manticore: {}
  piler_store: {}

and this is the configuration I found to connect ldap from zimbra

// authentication against an ldap directory (disabled by default)

$config['ENABLE_LDAP_AUTH'] = 1; 
$config['LDAP_HOST'] = 'mail.dominio.cu'; 
$config['LDAP_HELPER_DN'] = 'cn=....';
$config['LDAP_HELPER_PASSWORD'] = 'XXXXX'; 
$config['LDAP_MAIL_ATTR'] = 'mail';
$config['LDAP_AUDITOR_MEMBER_DN'] = '';
$config['LDAP_ADMIN_MEMBER_DN'] = '';
$config['LDAP_BASE_DN'] = '';

// AD specific settings
//
$config['LDAP_ACCOUNT_OBJECTCLASS'] = 'user';
$config['LDAP_DISTRIBUTIONLIST_OBJECTCLASS'] = 'group';
$config['LDAP_DISTRIBUTIONLIST_ATTR'] = 'member';
$config['LDAP_MAIL_ATTR'] = 'proxyAddresses';

// zimbra specific settings
$config['LDAP_HELPER_DN'] = 'uid=zimbra,cn=admins,cn=zimbra';
$config['LDAP_ACCOUNT_OBJECTCLASS'] = 'zimbraAccount';
$config['LDAP_DISTRIBUTIONLIST_OBJECTCLASS'] = 'zimbraDistributionList';
$config['LDAP_DISTRIBUTIONLIST_ATTR'] = 'zimbraMailForwardingAddress';
$config['LDAP_MAIL_ATTR'] = 'mail';

thanks for any help

**Comment by jsuto** [comment]
Well, the zimbra related stuff shall go to /etc/piler/config-site.php file. I can see that you have a volume for /etc/piler dir contents, so just edit that file on the docker volume, and you should be fine.

Btw. when using Zimbra, you don't need the AD specific config options.

**Reply by cubanlinux** [reply]
After several hours until 4am I managed to modify the file and add line
$config['ENABLE_LDAP_AUTH'] = 1;
more less an ldap section is activated on the website
but after adding the line nothing happens on the web even restarting the container

Is there a way to add that line
$config['ENABLE_LDAP_AUTH'] = 1;
 in the docker-compose.yml so when you build the container it is already in the file etc/piler/config-site.php

**Reply by jsuto** [reply, marked as answer]
But you can edit /etc/piler/config-site.php, it's there for you to add any local site specific settings.
You don't have to restart anything to apply the settings in config-site.php. If ldap is enabled, then try to login using a zimbra user account.

**Reply by cubanlinux** [reply]
In the end I managed to connect to the Windows ldap active directory right now with a clean installation in Ubuntu 24.04 and not with docker
with a script that uses your own github master

https://gist.github.com/gkwurst/87d240cf518cdbea8a78403c51f14885

after investigating each log and errors
I found that in the config-site.php I was having too many options and they conflicted
just leave the ones in the active directory and that's it 
Now I'm going to configure my Zimbra server so that it sends the copy to the piler

I just hope that after two sleepless nights I don't mess up the Zimbra mail server.

**Comment by jsuto** [comment]
Great! :-)
